llama_stack/apis/datatypes.py

```diff
@@ -42,9 +43,9 @@ def __call__(cls, value):
             if member_name in cls._member_map_:
                 return cls._member_map_[member_name]
 
-            # Instead of creating a new member, raise ValueError to force users to use Api.add() to
-            # register new APIs explicitly
-            raise ValueError(f"API '{value}' does not exist. Use Api.add() to register new APIs.") from e
+            # Instead of creating a new member, raise UnregisteredAPIError to force users to use
+            # Api.add() to register new APIs explicitly
+            raise UnregisteredAPIError(value) from e
```

Comment:
when is this encountered by the _user_ ? and the not the Stack? 

also I don't believe this error class is a good example for creating standard error classes. they should only be done for consistency in messages and that happens when there are at least 2-3 callsites for the same error. I cannot imagine it being used in more than one place.